Feature/notifications

mboskamp opened the pull request.

mboskamp committed:
Connect firebase to App

mboskamp committed:
Add message handling services

mboskamp committed:
Rename NotificationService Tag

mboskamp committed:
Merge remote-tracking branch 'origin/develop' into feature/notifications

mboskamp committed:
Merge branch 'develop' into feature/notifications

mboskamp committed:
Merge branch 'develop' into feature/notifications

mboskamp committed:
Open the specific Session mentioned in push notification when clicked on

elen-niedermeyer commented on app/src/main/java/com/quizarena/menu/MainMenuActivity.kt:
```diff
@@ -46,20 +47,32 @@ class MainMenuActivity : AppCompatActivity() {
         } else {
-            // activity was started manually, not by a link
+            val sessionID = intent.getStringExtra("session")
+            if(sessionID != null){
```
Ich komme hier in eine Endlosschleife. Wenn ich auf Zurück klicke, dann wird der gleiche Intent aufgerufen, der diesen Parameter gesetzt hat. Ich werde dann also direkt wieder zur DetailView geleitet. Das Extra des Intent muss auf null gesetzt werden. Ich denk, das sollte es dann lösen.

gleicher Grund warum ich die intent.action in Zeile 74 setze

mboskamp replied:
Ist geändert

elen-niedermeyer commented on app/src/main/java/com/quizarena/menu/MainMenuActivity.kt:
```diff
@@ -46,20 +47,32 @@ class MainMenuActivity : AppCompatActivity() {
+                // the activity was opened by a push notification
+                // go to the needed activity
+                val session = SessionSharing().getSessionFromID(this, sessionID)
+                startSessionDetailActivity(session)
```
Du solltest hier besser die SessionParticipantDetailActivity (oder wie die heißt) aufrufen. 
Ich gehe beim Sharing davon aus, dass der User noch nicht an der Session teilgenommen hat.
Hier gehen wir aber davon aus, dass er die Nachricht empfängt, wenn er teilgenommen hat. Das ist dann eine andere Activity.

Noch schöner wäre es natürlich, wenn du in der startSessionDetailView-Methode prüfst ob der Nutzer Teilnehmer ist (Ist im Session Objekt bereits als boolean hinterlegt) und dann die richtige Activity aufrufst.
Dann wäre es unabhängig von Sharing und Push und wir würden uns immer die Weiterleitung sparen :)

mboskamp replied:
Ich starte jetzt die jeweils passende Activity. Die Weiterleitungen habe ich noch nicht rausgenommen. Leider bekomme ich jetzt bei der letzten Frage eine IndexOutOfBoundsException die ich mir nicht erklären kann. Hat von euch jemand eine Idee?

mhoellner replied:
Wir haben einen IndexOutOfBounds-Fehler, der bei mir aufgetreten ist entfernen können.
Merge doch mal den aktuellen develop in dein Feature und guck ob es dann klappt.

elen-niedermeyer replied:
Die Weiterleitung würde ich zur Sicherheit auch drin lassen. Man weiß ja nie, von wo das vielleicht noch aufgerufen wird...

mboskamp replied:
Funktioniert jetzt ohne Probleme. Kein Fehler mehr. 👍

mboskamp committed:
Prevent forwarding when returning from SessionDetail

mboskamp committed:
Use correct activity for push notifications

We can use SessionParticipantDetailActivity for push notifications as those will only be sent to participants
